src/Elan.Web/Chat/ChatHub.cs

```diff
@@ -13,22 +17,35 @@ public class ChatHub: Hub
     {
         private readonly IUserService _userService;
         private readonly IChatService _chatService;
+        private readonly IHubContext<NotificationHub> _notificationHub;
+        private readonly INotificationService _notificationService;
 
         private readonly object _lock = new object();
 
         private static readonly Dictionary<string, string> _connections = new Dictionary<string, string>();
 
-        public ChatHub(IUserService userService, IChatService chatService)
+        public ChatHub(IUserService userService,
+                       IChatService chatService,
+                       INotificationService notificationService,
+                       IHubContext<NotificationHub> notificationHub)
         {
             _userService = userService;
             _chatService = chatService;
+            _notificationService = notificationService;
+            _notificationHub = notificationHub;
         }
 
         public async Task SendMessage(string toUserId, string message)
         {
             var userFrom = await _userService.GetUserByName(Context.User.Identity.Name);
             var userTo = await _userService.GetUserById(toUserId);
-            
+
+            if(!(await _notificationService.HasUnreadChatNotificationWithUSer(userTo, userFrom)))
```

Comment:
move value to variable: var hasUnreadChatNotification = await _notificationService.HasUnreadChatNotificationWithUSer(userTo, userFrom) then inverse if condition. Move sending notifications to the same try statement as sending chat message (no message -> no notification)